src/main/scala/com/zeotap/data/io/source/spark/constructs/DataFrameOps.scala

```diff
@@ -71,6 +75,58 @@ object DataFrameOps {
 
       dataFrame.withColumn("inputPathColumn", addInputPathColumn(input_file_name())).select("inputPathColumn").distinct().collect.map(row => row.getString(0))
     }
+
+    /**
+     * Partitions given dataframe into number of partitions provided and writes the
+     * partitioned dataframe to the intermediate path provided (partitioning happens only if the intermediate data is not already present),
+     * so that parallel processing can take place.
+     *
+     * @param numberOfPartitions         is the number of partitions we want in the dataframe.
+     * @param intermediatePath           is the intermediate path in which the partitioned data frame is written.
+     * @param prioritiseIntermediatePath is boolean value which denotes whether the intermediate path (i.e already partitioned path)
+     *                                   needs to prioritised or should we force repartition. If it is true then data at intermediate path
+     *                                   will be returned(if non empty).
+     * @return Returns Dataframe with specified number of partitions.
+     */
+
+    def distributedLoad(numberOfPartitions: Option[Int], intermediatePath: String, prioritiseIntermediatePath: Option[Boolean]): DataFrame = {
+      val valPrioritiseIntermediatePath = prioritiseIntermediatePath match {
+        case Some(priority) => priority
+        case None => true
+      }
+
+      val spark = dataFrame.sparkSession
+      if (!valPrioritiseIntermediatePath) {
+        defaultSplit(numberOfPartitions, intermediatePath)
+      }
+      else {
+        val intermediateDf = util.Try {
+          ParquetSparkLoader().load(intermediatePath).buildUnsafe(spark)
+        }
+
+        intermediateDf match {
+          case Success(df) => df
+          case Failure(exception) =>
+            logger.log.info(s"Failed to load data from $intermediatePath " + exception.getMessage + " proceeding with default split strategy!")
+            defaultSplit(numberOfPartitions, intermediatePath)
+        }
+      }
+    }
+
+    /*
+    Default Splitting strategy, takes Raw Input Dataframe, re-partitions it, writes the partitioned dataframe
+    to intermediate path, loads the dataframe present at intermediate path and returns it for further processing.
+     */
+    def defaultSplit(numberOfPartitions: Option[Int], intermediatePath: String): DataFrame = {
+      val spark = dataFrame.sparkSession
+      val valNumberOfPartitions = numberOfPartitions match {
+        case Some(partitions) => partitions
+        case None => 200
+      }
+      val partitionedDf = dataFrame.repartition(valNumberOfPartitions)
+      ParquetSparkWriter().addSaveMode(Overwrite).save(intermediatePath).buildUnsafe(partitionedDf)
+      ParquetSparkLoader().load(intermediatePath).buildUnsafe(spark)
+    }
```

Comment:
I feel that these features should be in the DataFrameReaderOps and not DataFrameOps